ambry-api/src/test/java/com.github.ambry/router/GetBlobOptionsTest.java

```diff
@@ -75,21 +90,40 @@ public void testToStringEqualsAndHashcode() {
     GetBlobOptions b = new GetBlobOptionsBuilder().operationType(type).getOption(getOption).range(byteRange).build();
     assertEquals("GetBlobOptions should be equal", a, b);
     assertEquals("GetBlobOptions hashcodes should be equal", a.hashCode(), b.hashCode());
-    assertEquals("toString output not as expected",
-        "GetBlobOptions{operationType=" + type + ", getOption=" + getOption + ", range=" + byteRange.toString() + "}",
-        a.toString());
+    assertEquals("GetBlobOptions toString should be equal", a.toString(), b.toString());
 
-    b = new GetBlobOptionsBuilder().operationType(type)
-        .getOption(getOption)
-        .range(ByteRange.fromOffsetRange(2, 7))
-        .build();
-    assertThat("GetBlobOptions should not be equal.", a, not(b));
+    // Change OperationType
     b = new GetBlobOptionsBuilder().operationType(GetBlobOptions.OperationType.All)
         .getOption(getOption)
         .range(byteRange)
         .build();
-    assertThat("GetBlobOptions should not be equal.", a, not(b));
+    assertObjectsAreDistinct(a, b);
+
+    // Change GetOption
     b = new GetBlobOptionsBuilder().operationType(type).getOption(GetOption.Include_All).range(byteRange).build();
+    assertObjectsAreDistinct(a, b);
+
+    // Change range
+    b = new GetBlobOptionsBuilder().operationType(type)
+        .getOption(getOption)
+        .range(ByteRange.fromOffsetRange(2, 7))
+        .build();
+    assertObjectsAreDistinct(a, b);
+
+    // Change rawMode (need to omit range)
+    a = new GetBlobOptionsBuilder().operationType(type).getOption(getOption).build();
+    b = new GetBlobOptionsBuilder().operationType(type).getOption(getOption).rawMode(true).build();
+    assertObjectsAreDistinct(a, b);
+  }
+
+  /**
+   * Verify that two instances of GetBlobOptions are not equal and have distinct hashcodes and toStrings.
+   * @param a first instance
+   * @param b second instance
+   */
+  private static void assertObjectsAreDistinct(GetBlobOptions a, GetBlobOptions b) {
     assertThat("GetBlobOptions should not be equal.", a, not(b));
+    assertThat("GetBlobOptions hashcodes should not be equal", a.hashCode(), not(b.hashCode()));
```

Comment:
Technically that's true, but the chance of collision is 1 in several billion.
Is this class ever used as a hash key?  I was wondering why it even needed to define a hashcode.

Reply:
This class is not likely to be used as a hashkey (unless some tests do that?). It is mostly there since we were implementing an equals method anyways.